conda_forge_tick/mamba_solver.py

```diff
@@ -297,8 +364,114 @@ def solve(self, specs) -> Tuple[bool, List[str]]:
+            for _, _, jdata in to_link:
+                data = json.loads(jdata)
+                solution.append(
+                    " ".join([data["name"], data["version"], data["build"]]),
+                )
+
+            if get_run_exports:
+                run_exports = self._get_run_exports(to_link)
+
+        if get_run_exports:
+            return success, err, solution, run_exports
+        else:
+            return success, err, solution
+
+    def _get_run_exports(self, link_tuples):
+        """Given tuples of (channel, file, json repodata shard) produce a
+        dict with the weak and strong run exports for the packages.
+        """
 
-        return success, err
+        with ProcessPoolExecutor(max_workers=None) as exe:
+            futures = []
+            for link_tuple in link_tuples:
+                if link_tuple not in self.run_exports:
+                    cd = download_channeldata(link_tuple[0].rsplit("/", maxsplit=1)[0])
+                    data = json.loads(link_tuple[2])
+                    name = data["name"]
+                    version = data["version"]
+                    if cd.get("packages", {}).get(name, {}).get("run_exports", {}):
+                        # channel data run exports
+                        cd_rx = (
+                            cd.get("packages", {}).get(name, {}).get("run_exports", {})
+                        )
+
+                        # libcfgraph location
+                        if link_tuple[1].endswith(".tar.bz2"):
+                            pkg_nm = link_tuple[1][: -len(".tar.bz2")]
+                        else:
+                            pkg_nm = link_tuple[1][: -len(".conda")]
+                        channel_subdir = "/".join(link_tuple[0].split("/")[-2:])
+                        libcfg_pth = (
+                            f"../libcfgraph/artifacts/{name}/"
+                            f"{channel_subdir}/{pkg_nm}.json"
+                        )
+
+                        if os.path.exists(libcfg_pth):
+                            with open(libcfg_pth) as fp:
+                                data = json.load(fp)
+
+                            rx = (
+                                data.get("rendered_recipe", {})
+                                .get("build", {})
+                                .get("run_exports", {})
+                            )
+                            if rx:
+                                self.run_exports[link_tuple] = copy.deepcopy(
+                                    DEFAULT_RUN_EXPORTS,
+                                )
+                                for k in rx:
+                                    if k in DEFAULT_RUN_EXPORTS:
+                                        for _k in rx[k]:
+                                            self.run_exports[link_tuple][k].add(_k)
+                                    else:
+                                        self.run_exports[link_tuple]["weak"].add(k)
```

Comment:
Isn't `weak` in default run exports?

Reply:
This else captures syntax like

```yaml
build:
  run_exports:
    - blah
```

which makes `blah` a weak run export by default.

Reply:
we support that? oof

Reply:
yup it is: https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#export-runtime-requirements

conda-build might always put out an explicit weak section, but I didn't know so I added this just in case.